### DISCUSSION TITLE
REST Query results in data provider

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#12905] [Help] [answered]
Hi all,

is there a way to use the results of a REST Query in a data provider?

My query:
https://api.cognitive.microsofttranslator.com/languages?api-version=3.0&scope=translation

Sincerely,
Kevin

**Comment by ConorWebb96** [comment]
Hey @kasgithub8,

If you use the [rest connector](https://docs.budibase.com/docs/rest) and import the above query you can then send and save. From here you should then be able to select this query to be used inside a data provider.
<img width="261" alt="Screenshot 2024-01-30 at 09 02 42" src="https://github.com/Budibase/budibase/assets/126772285/17c032e4-88ba-457d-98be-2e85f5449a3e">
<img width="313" alt="Screenshot 2024-01-30 at 09 03 00" src="https://github.com/Budibase/budibase/assets/126772285/2b969201-7459-44c1-be69-adafc818aec6">

Hopefully this helps.

**Reply by kasgithub8** [reply]
Right, that's how far I got.
The result is the usual [object Object] and when I try to use it in a options picker there's only one available field, "translation" which is the array outside of the answers I want to use.

**Reply by ConorWebb96** [reply]
The array you are trying to use is structured in a weird way.  Its not a normal occurrence for object keys to be named. To get this to work in a select, all you would need to do is create a new array that only outputs the object key names. Then return this newly structured array which should now be suitable to be used in the multi-select component.

Transformer js here
```js
const languageCodesArray = Object.keys(data.translation);

return languageCodesArray;
```

Heres screenshots of this process.
<img width="1109" alt="Screenshot 2024-01-30 at 10 09 23" src="https://github.com/Budibase/budibase/assets/126772285/924fe5b3-796a-4d94-b303-1b4ac5ac5234">
<img width="1115" alt="Screenshot 2024-01-30 at 10 09 31" src="https://github.com/Budibase/budibase/assets/126772285/3385e2c0-bff8-489a-a5bf-f5b4490f4125">
<img width="1381" alt="Screenshot 2024-01-30 at 10 09 39" src="https://github.com/Budibase/budibase/assets/126772285/0f31c762-e701-402b-a6f8-4b7aab733016">
<img width="1381" alt="Screenshot 2024-01-30 at 10 10 43" src="https://github.com/Budibase/budibase/assets/126772285/3efc6a6c-f3c6-4753-b7e4-6a4f4a3f0bdd">

**Reply by kasgithub8** [reply]
@ConorWebb96 ah! That's a cool use case for the transformer. And yes, the array is weirdly structured like that because it's Microsoft 😄.
My end goal is to translate text with this query: `https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&from=de&to=bg`. 

Now my problem is that the Microsoft API only accepts the content-type: application/JSON.
When I fill the body with raw(JSON) `[
	{"text": "Hey, what's your name?"}
]` it works as expected. But you can't use the input of a text field because it's just raw JSON without any bindings. 
Do you know anything about that?

**Comment by kasgithub8** [comment, marked as answer]
Got it myself!
